odoo/addons/base/models/ir_mail_server.py

```diff
@@ -315,7 +315,8 @@ def connect(self, host=None, port=None, user=None, password=None, encryption=Non
             local, at, domain = smtp_user.rpartition('@')
             if at:
                 smtp_user = local + at + idna.encode(domain).decode('ascii')
-            connection.login(smtp_user, smtp_password or '')
+            mail_server = mail_server or self.env['ir.mail_server']
```

Comment:
Would move it as en else after ``elif not host: mail_server = self.sudo().search([], order='sequence', limit=1)`` that way mail_server is set once for all.

Reply:
`_find_mail_server` can return None in some cases, but I can move it above without an else